src/silx/gui/plot/items/core.py

```diff
@@ -658,9 +658,21 @@ def _setColormappedData(
         # Fill-up colormap range cache if values are provided
         if max_ is not None and numpy.isfinite(max_):
             if min_ is not None and numpy.isfinite(min_):
-                self.__cacheColormapRange[Colormap.LINEAR, Colormap.MINMAX] = min_, max_
+                if (Colormap.LINEAR, Colormap.MINMAX) not in self.__cacheColormapRange:
+                    self.__cacheColormapRange = {}
```

Comment:
This really the bad thing about the PR. We now need an extra level for the saturation.
Having `self.__cacheColormapRange[normalization, autoscalemode, saturation] ` would also make sense.
But in both cases the cache could quickly become heavy (we can limit it of course to n elements). But the calculation is also taking a bit of time...

Reply:
Yes it's true,you have to use a LRU cache.

If it's not already there you can add `lru-dict` to the dependencies.

Also, in this case i think you better to have 2 caches.

- One with the fixed values (the actual one)
- And the one with the infinite amount of values which have to be the LRU cache.

This way the fixed `MINMAX` values will not be dropped. I think that's important at some places. I can be wrong.

Another solution could be to use integer only for the saturation (do you really care about the difference between 5.0 and 5.1?)

Reply:
IMO we can leave with integers. This is the API provided by the GUI